addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3786,6 +3786,9 @@ var SnippetsMenu = Widget.extend({
      * @private
      */
     _onActivateSnippet: function (ev) {
+        if (ev.data.reloadTemplates) {
+            this._loadSnippetsTemplates();
+        }
```

Comment:
> Remark: there is already an implementation preventing sections with
> hidden widgets from being visible (see: `web_editor` >
> `updateOptionsUIVisibility()`). Despite this, the section remains
> visible since it contains `we-top-button-group` options.

Should not we fix that instead :point_up: ? -> "If all options are invisible, the section should not be visible, nor have a related overlay"?

Reloading "snippets template" is a costly operation as it destroys all options already instantiated. We should avoid doing it. More than that, I think the original code is the way it is supposed to be given the current editor options organization: "All column can control their padding, but only when inside a grid" => if we switch to grid, no new option is instantiated, it's just made visible". In general, we know "this item could have this option if ...", in that case, using option/widget visibility seems the way to go at the moment. It's like parallax option: most snippet do not use parallax, but all have the option... which only becomes visible when you add a background image. It would be annoying to do "now that we added a background image, reload all options in case an option target sections which have a background-image".

Maybe there is a better system/organization to consider in master. In stable, I think the visibility stuff should be fixed instead of this :+1:

Reply:
> Should not we fix that instead ☝️ ? -> "If all options are invisible, the section should not be visible, nor have a related overlay"?

@qsm-odoo Yes, I previously quickly checked the code to be sure there was no better solution (actually, that's why I mentioned the remark about `updateOptionsUIVisibility()` in the commit), but I thought not loading the snippets would lead to some inconsistencies... I adapted the code to simply hide empty `SnippetEditor`s and their overlays. I also updated the commit message to explain the diff. Just let me know if something needs to be adjusted here :+1: